lib/serializer.ex

```diff
@@ -48,26 +58,47 @@ defmodule Membrane.Element.IVF.Serializer do
 
   @impl true
   def handle_process(:input, buffer, ctx, state) do
-    %Buffer{payload: frame, metadata: %{timestamp: timestamp}} = buffer
+    %Buffer{payload: frame, pts: timestamp} = buffer
 
     ivf_frame =
       IVF.Headers.create_ivf_frame_header(byte_size(frame), timestamp, state.timebase) <>
         frame
 
     ivf_file_header =
-      if state.first_frame,
-        do:
-          IVF.Headers.create_ivf_header(
-            state.width,
-            state.height,
-            state.timebase,
-            state.frame_count,
-            ctx.pads.input.caps
-          )
+      if state.first_frame do
+        if state.width == 0 or state.height == 0,
+          do:
+            IO.warn("Serializing stream to IVF without width or height given via options or caps")
+
+        IVF.Headers.create_ivf_header(
+          state.width,
+          state.height,
+          state.timebase,
+          state.frame_count,
+          ctx.pads.input.caps
+        )
+      end
 
     ivf_buffer = (ivf_file_header || "") <> ivf_frame
 
     {{:ok, buffer: {:output, %Buffer{buffer | payload: ivf_buffer}}, redemand: :output},
      %State{state | first_frame: false}}
   end
+
+  defp update_state(state, new_fields) do
```

Comment:
`update_state` and `new_fields` are not really descriptive and self-explanatory, we need to find some better names for them. Maybe `apply_caps_to_state` or `update_state_with_caps`, and `new_field` -> `caps`.